### PR TITLE
Add Telugu language

### PR DESCRIPTION
**Pull request by kranthilakum**
A pull request to add the [Telugu language (ISO 639-1 `te`, ISO 639-2 `tel`)](https://en.wikipedia.org/wiki/Telugu_language) to date-fns library.
It is the third most spoken Indian language in India, [spoken by 82 million people](https://www.ethnologue.com/language/tel)

The language is already supported by the other libraries, e.g. 
- Moment.js [supported](https://github.com/moment/moment/blob/develop/locale/te.js)
- Day.js [supported](https://github.com/iamkun/dayjs/blob/dev/src/locale/te.js)

This is my first pull request to date-fns library. I have followed the i18n guidelines in [i18nContributionGuide.md](https://github.com/date-fns/date-fns/blob/master/docs/i18nContributionGuide.md) and [contribution guide](https://github.com/date-fns/date-fns/blob/master/CONTRIBUTING.md). Please review.

### CODE REVIEW

File: src/locale/te/_lib/localize/index.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+// CLDR #1613 - #1628
+var quarterValues = {
+  narrow: ['1', '2', '3', '4'],
+  abbreviated: ['త్రై1', 'త్రై2', 'త్రై3', 'త్రై4'],
+  wide: ['1వ త్రైమాసికం', '2వ త్రైమాసికం', '3వ త్రైమాసికం', '4వ త్రైమాసికం']
+}
+
+// CLDR #1637 - #1708
+// Note: in English, the names of days of the week and months are capitalized.
```
Comment (dkozickis) [inline src/locale/te/_lib/localize/index.js]: You can remove this remark and leave remark specific to Telugu (if any)

File: src/locale/te/_lib/localize/index.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    afternoon: 'మధ్యాహ్నం',
+    evening: 'సాయంత్రం',
+    night: 'రాత్రి'
+  }
+}
+
+function ordinalNumber(dirtyNumber, _dirtyOptions) {
+  var number = Number(dirtyNumber)
+
+  // If ordinal numbers depend on context, for example,
```
Comment (dkozickis) [inline src/locale/te/_lib/localize/index.js]: Please remove this comment section.

File: src/locale/te/index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * @type {Locale}
+ * @category Locales
+ * @summary Telugu locale
+ * @language Telugu
+ * @iso-639-2 tel
+ * @author Kranthi Lakum [@kranthilakum]{@link https://github.com/kranthilakum}
+ */
+
```
Comment (dkozickis) [inline src/locale/te/index.js]: Please remove blank line between JSDoc and `locale`

**Comment by dkozickis** [comment]
Thank you for your contribution @kranthilakum ! 🙇 

I've left some minor comments, please address them. Otherwise this one seems good to go @leshakoss ?

File: src/locale/te/_lib/localize/index.js (hunk 1 above)
Comment (kranthilakum) [reply src/locale/te/_lib/localize/index.js]: Removed the remark 5d5c1ef45

File: src/locale/te/_lib/localize/index.js (hunk 2 above)
Comment (kranthilakum) [reply src/locale/te/_lib/localize/index.js]: Removed the comment section 5d5c1ef45

File: src/locale/te/index.js (hunk 3 above)
Comment (kranthilakum) [reply src/locale/te/index.js]: Removed the blank line 5d5c1ef45

**Comment by dkozickis** [comment]
Thank you for your contribution @kranthilakum!

**Pull request merged**